symengine/cwrapper.cpp

```diff
@@ -386,4 +386,11 @@ size_t basic_hash(const basic self)
     return self->m->hash();
 }
 
+void basic_subs2(basic s, const basic e, const basic a, const basic b)
+{
+    SymEngine::map_basic_basic temp;
+    temp.insert(std::pair< RCP<const Basic>, RCP<const Basic> >(a->m,b->m));
```

Comment:
It can be use in other places as well. A map initialized to `{{a, b}}, {c,d}}` is a map with `a` mapping to `b` and `c` mapping to `d`